packages/flutter_tools/test/general.shard/build_system/targets/linux_test.dart

```diff
@@ -20,33 +20,48 @@ import '../../../src/common.dart';
 import '../../../src/context.dart';
 
 void main() {
-  testWithoutContext('Copies files to correct cache directory, excluding unrelated code', () async {
-    final FileSystem fileSystem = MemoryFileSystem.test();
-    final Artifacts artifacts = Artifacts.test();
-    setUpCacheDirectory(fileSystem, artifacts);
-
-    final Environment testEnvironment = Environment.test(
-      fileSystem.currentDirectory,
-      defines: <String, String>{
-        kBuildMode: 'debug',
-      },
-      artifacts: artifacts,
-      processManager: FakeProcessManager.any(),
-      fileSystem: fileSystem,
-      logger: BufferLogger.test(),
-    );
-    testEnvironment.buildDir.createSync(recursive: true);
+  final List<TargetPlatform> targetPlatforms = <TargetPlatform>[
+    TargetPlatform.linux_arm64,
+    TargetPlatform.linux_arm64
+  ];
 
-    await const UnpackLinux().build(testEnvironment);
+  testWithoutContext('Copies files to correct cache directory, excluding unrelated code', () async {
```

Comment:
Its a little unclear what this is testing - and the test description won't help if there is an issue on x64 vs arm. Instead of using a loop in the test body, could you create two separate test cases? Its OK if the test bodies are a bit copy-pasted, generally I would prefer tests that are a bit verbose with simpler logic.

Reply:
Similar comments in many of the other test cases

Reply:
Sorry, I thought that it was basically OK if the existing tests that used `linux-x64` passed with `linux-arm64`. Now, I fixed all unit tests.